### PR TITLE
[#140832481] Update the product page route

### PR DESCRIPTION
**Pull request by paroxp**
## What

We're renaming the application we're hosting. In order to ensure, it shan't go down, we'll deploy it under a new name and route.

The old application will be deleted, upon merging of this commit, therefore we need to ensure CloudFront will be reading from a new route straight away.

## How to review

Sanity check should be sufficient enough.
Make sure, this is deployed AFTER the new application is up and running.

## Who can review

Anyone but myself.

### CODE REVIEW

File: terraform/cloudfront/instances.tf [hunk 1]
```diff
@@ -9,7 +9,7 @@ module "cloudfront_paas_product_page" {
 
   name    = "PaaS Product Page"
   aliases = ["www.${var.system_dns_zone_name}"]
-  origin  = "govuk-paas.cloudapps.digital"
+  origin  = "paas-product-page.cloudapps.digital"
   comment = "Serve the govuk-paas under the gov.uk domain."
```
Comment (dcarley) [inline terraform/cloudfront/instances.tf]: Rename `the govuk-paas` to `paas-product-page`?

**Comment by dcarley** [comment]
> The old application will be deleted, upon merging of this commit, therefore we need to ensure CloudFront will be reading from a new route straight away.

Am I right in thinking that this instruction has changed since the PR was first opened, because we're now deploying another app in parallel? Two further questions if so..

1. Should I delete the old `govuk-paas` app after this is deployed?
1. Will requests for `paas-roadmap` be served to both the new and old apps up until the old app is deleted?

**Comment by paroxp** [comment]
It may be me trying to hard...

So, by the time you'll come to this PR, the new application route should be up and running. If by a slight chance, this PR gets merged and deployed on production, CloudFront may want to access the new URL which doesn't yet exist, causing itself to cache the 404 page.

The old application, should be deleted when we're sure, that the CloudFront is now reading from a new URL.

> Will requests for paas-roadmap be served to both the new and old apps up until the old app is deleted?

Most likely, CF will throw an exception, saying that the route is already in use. To avoid that, we'll need to remove this route (`paas-roadmap`) from the old app before the new one will get pushed. Not ideal, but I'm not sure, where and how is this route even used.

**Comment by dcarley** [comment]
Both apps have the same route in my dev environment:
```
paas-product-page   started           2/2         64M      1G     paas-roadmap.dcarley.dev.cloudpipelineapps.digital, paas-product-page.dcarley.dev.cloudpipelineapps.digital
govuk-paas          started           2/2         64M      1G     govuk-paas.dcarley.dev.cloudpipelineapps.digital, paas-roadmap.dcarley.dev.cloudpipelineapps.digital
```

I think CF will distribute traffic evenly between them until the old app is deleted. It's essentially the same trick that the blue-green plugin uses. So it should be fine.

The reason for the route is described in alphagov/paas-product-page@9b57c6d6a5a0fce83c9b59ba93ea617f0a8f4117

**Comment by dcarley** [comment]
The new app is deployed. I'm happy that it seems to be working:
```
➜  paas-cf git:(master) ✗ curlh https://paas-roadmap.cloudapps.digital
HTTP/1.1 301 Moved Permanently
Content-Length: 178
Content-Type: text/html
Date: Wed, 29 Mar 2017 13:40:04 GMT
Location: https://www.cloud.service.gov.uk/roadmap.html
Server: nginx
X-Vcap-Request-Id: f32409a8-2a48-4e91-5fe1-711d389b7267
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload

➜  paas-cf git:(master) ✗ curlh https://paas-product-page.cloudapps.digital
HTTP/1.1 200 OK
Accept-Ranges: bytes
Content-Length: 11504
Content-Type: text/html; charset=utf-8
Date: Wed, 29 Mar 2017 13:40:09 GMT
Etag: "58dbb888-2cf0"
Last-Modified: Wed, 29 Mar 2017 13:37:12 GMT
Server: nginx
Vary: Accept-Encoding
X-Vcap-Request-Id: 599d659e-5225-4af3-72c4-d9c2ecd97ffa
Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
```

Requests were served by the new app:
```
➜  paas-cf git:(master) ✗ cf logs paas-product-page
Connected, tailing logs for app paas-product-page in org govuk-paas / space docs as dan.carley@digital.cabinet-office.gov.uk...

2017-03-29T14:40:04.51+0100 [RTR/1]      OUT paas-roadmap.cloudapps.digital - [2017-03-29T13:40:04.516+0000] "GET / HTTP/1.1" 301 0 178 "-" "curl/7.43.0" "127.0.0.1:53016" "10.0.34.8:62158" x_forwarded_for:"80.194.77.100, 127.0.0.1" x_forwarded_proto:"https" vcap_request_id:"f32409a8-2a48-4e91-5fe1-711d389b7267" response_time:0.001678321 app_id:"8e7f45da-4fae-46d0-88e5-a81dee677527" app_index:"0"
2017-03-29T14:40:09.99+0100 [RTR/1]      OUT paas-product-page.cloudapps.digital - [2017-03-29T13:40:09.989+0000] "GET / HTTP/1.1" 200 0 11504 "-" "curl/7.43.0" "127.0.0.1:53274" "10.0.32.8:62579" x_forwarded_for:"80.194.77.100, 127.0.0.1" x_forwarded_proto:"https" vcap_request_id:"599d659e-5225-4af3-72c4-d9c2ecd97ffa" response_time:0.002188874 app_id:"8e7f45da-4fae-46d0-88e5-a81dee677527" app_index:"1"
```

**Comment by dcarley** [comment]
I realised that this doesn't deploy automatically. So I've run:
```
make prod setup_cdn_instances ACTION=plan
…
make prod setup_cdn_instances ACTION=apply
```

I then verified that the old app wasn't receiving any traffic with `cf logs` and deleted it:
```
➜  paas-cf git:(master) ✗ cf delete govuk-paas

Really delete the app govuk-paas?> yes
Deleting app govuk-paas in org govuk-paas / space docs as dan.carley@digital.cabinet-office.gov.uk...
OK
```

I deleted the orphaned route separately, because `cf delete govuk-paas -f` would have removed the `govuk-roadmap` route from the new app:
```
➜  paas-cf git:(master) ✗ cf delete-route cloudapps.digital --hostname govuk-paas -f
Deleting route govuk-paas.cloudapps.digital...
OK
```

**Pull request merged**